src/imagery/i.eodag/i.eodag.py

```diff
@@ -300,82 +396,199 @@ def no_fallback_search(search_parameters, provider):
+            search_result = search_result.filter_overlap(
+                geometry=geometry, contains=True
+            )
+        # Product's geometry is within the AOI
+        elif area_relation == "IsWithin":
+            search_result = search_result.filter_overlap(geometry=geometry, within=True)
+        # Percentage of area covered from the AOI by the product's geometry
+        elif minimum_overlap:
+            search_result = search_result.filter_overlap(
+                geometry=geometry, minimum_overlap=int(minimum_overlap)
+            )
 
-def apply_filters(search_result):
-    filtered_result = []
-    for product in search_result:
-        valid = True
-        if (
-            options["clouds"]
-            and "cloudCover" in product.properties
-            and product.properties["cloudCover"] is not None
-            and product.properties["cloudCover"] > int(options["clouds"])
-        ):
-            valid = False
-        if valid:
-            filtered_result.append(product)
-    return filtered_result
+    if cloud_cover:
+        search_result = search_result.filter_property(
+            operator="le", cloudCover=int(cloud_cover)
+        )
+
+    search_result = search_result.filter_date(start=start_date, end=end_date)
+    search_result = remove_duplicates(search_result)
+
+    postfilter_count = len(search_result)
+    gs.verbose(
+        _("{} product(s) filtered out.".format(prefilter_count - postfilter_count))
+    )
+
+    return search_result
+
+
+def sort_result(search_result):
+    """Sorts search results according to options['sort'] and options['order']
+
+    options['sort'] parameters and options['order'] are matched correspondingly.
+    If options['order'] parameters are not suffcient,
+    'asc' will be used by default.
+
+    :param search_result: EO products to be sorted
+    :type search_result: class'eodag.api.search_result.SearchResult'
+
+    :return: Sorted EO products
+    :rtype: class:'eodag.api.search_result.SearchResult'
+    """
+    gs.verbose(_("Sorting..."))
+
+    sort_keys = options["sort"].split(",")
+    sort_order = options["order"].split(",")
+    sort_order.extend(["asc"] * max(0, len(sort_keys) - len(sort_order)))
+
+    # Sort keys and sort orders are matched respectively
+    def products_compare(first, second):
+        for idx, sort_key in enumerate(sort_keys):
+            if sort_key == "ingestiondate":
+                first_value = first.properties["startTimeFromAscendingNode"]
+                second_value = second.properties["startTimeFromAscendingNode"]
+            elif sort_key == "cloudcover":
+                first_value = first.properties["cloudCover"]
+                second_value = second.properties["cloudCover"]
+            if first_value < second_value:
+                return 1 if sort_order[idx] == "desc" else -1
+            elif first_value > second_value:
+                return -1 if sort_order[idx] == "desc" else 1
```

Comment:
What is supposed to happen if `first_value > second_value`?

After a return elif is unnecessary. Could be removed
```suggestion
            return -1 if sort_order[idx] == "desc" else 1
```
unless you want to handle first_value > second_value specifically...

Reply:
If the values are equal it should return 0, that is why the second **elif** is necessary.